src/main/java/org/dita/dost/writer/SeparateChunkTopicParser.java

```diff
@@ -293,11 +295,28 @@ public void startElement(final String uri, final String localName, final String
             throws SAXException {
         final String cls = atts.getValue(ATTRIBUTE_NAME_CLASS);
         final String id = atts.getValue(ATTRIBUTE_NAME_ID);
+        final AttributesImpl attsMod = new AttributesImpl(atts);
+        final String xmlLang = atts.getValue(ATTRIBUTE_NAME_XML_LANG);
+        final String currentLang;
+
 
         if (skip && skipLevel > 0) {
             skipLevel++;
         }
 
+        if (xmlLang != null) {
+            currentLang = xmlLang;
+        }
+        else {
+            if (lang.size() > 0) {
+                currentLang = lang.peek();
+            }
+            else {
+                currentLang = "";
```

Comment:
[Null elements are prohibited](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayDeque.html) in ArrayDeque. Using null is is likely to complicate the code.